tests/FSharp.Compiler.ComponentTests/TypeChecks/Graph/Scenarios.fs

```diff
@@ -1042,4 +1042,31 @@ type Bar(foo: MyRootNamespace.A.Foo, s: string) = class end
 """
                     (set [| 0 |])
             ]
+        scenario
+            "Library with using global namespace"
+            [
+                sourceFile
+                    "Library.fs"
+                    """
+namespace Lib
+module File1 =
+    let mutable discState = System.DateTime.Now
+
+module File2 =
+    [<Struct>]
+    type DiscState(rep : int) =
+        member this.Rep = rep
+
+    let mutable discState = DiscState(0)
+                    """
+                    Set.empty
+                sourceFile
+                    "App.fs"
+                    """
+module X
+let v = global.Lib.File1.discState.Second
+let v2 = global.Lib.File2.discState.Rep
+                    """
+                    Set.empty
```

Comment:
This looks wrong. The second file needs information from the first file, so that should be `set [| 0 |]` instead.

Reply:
Wait, I actually have the change locally, didn't push it.